fenixedu-ist-giaf-invoices/src/main/java/pt/ist/fenixedu/giaf/invoices/task/FindThirdPartyReciepts.java

```diff
@@ -0,0 +1,220 @@
+package pt.ist.fenixedu.giaf.invoices.task;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.Properties;
+
+import javax.activation.DataHandler;
+import javax.activation.DataSource;
+import javax.mail.Message;
+import javax.mail.MessagingException;
+import javax.mail.Multipart;
+import javax.mail.Session;
+import javax.mail.Transport;
+import javax.mail.internet.AddressException;
+import javax.mail.internet.InternetAddress;
+import javax.mail.internet.MimeBodyPart;
+import javax.mail.internet.MimeMessage;
+import javax.mail.internet.MimeMultipart;
+import javax.mail.util.ByteArrayDataSource;
+
+import org.fenixedu.academic.FenixEduAcademicConfiguration;
+import org.fenixedu.academic.domain.Country;
+import org.fenixedu.academic.domain.ExecutionYear;
+import org.fenixedu.academic.domain.Person;
+import org.fenixedu.academic.domain.accounting.AccountingTransaction;
+import org.fenixedu.academic.domain.accounting.AccountingTransactionDetail;
+import org.fenixedu.academic.domain.accounting.Entry;
+import org.fenixedu.academic.domain.accounting.Event;
+import org.fenixedu.academic.domain.accounting.Receipt;
+import org.fenixedu.academic.domain.util.email.Sender;
+import org.fenixedu.bennu.GiafInvoiceConfiguration;
+import org.fenixedu.bennu.core.domain.Bennu;
+import org.fenixedu.bennu.core.domain.User;
+import org.fenixedu.bennu.scheduler.custom.CustomTask;
+import org.fenixedu.commons.spreadsheet.Spreadsheet;
+import org.fenixedu.commons.spreadsheet.Spreadsheet.Row;
+import org.joda.time.DateTime;
+
+import com.google.common.base.Objects;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+import pt.ist.fenixedu.giaf.invoices.DebtCycleType;
+import pt.ist.fenixedu.giaf.invoices.Utils;
+import pt.ist.fenixframework.FenixFramework;
+
+public class FindThirdPartyReciepts extends CustomTask {
+
+    private static final String DIR = GiafInvoiceConfiguration.getConfiguration().giafInvoiceDir();
+
+    @Override
+    public void runTask() throws Exception {
+        final Spreadsheet sheet = new Spreadsheet("Invoices");
+
+        process(sheet, new File(DIR));
+
+        final ByteArrayOutputStream stream = new ByteArrayOutputStream();
+        sheet.exportToXLSSheet(stream);
+
+        output("invoices.xls", stream.toByteArray());
+
+        final String subject = "Recibos passados a entidades terceiras";
+        final String body = "Listagem atualizada com os 'recibos' passados no Fénix entidades terceiras: " + new DateTime().toString("yyyy-MM-dd HH:mm");
+
+        TaskUtils.sendErrorReport("invoices_" + new DateTime().toString("yyyy_MM_dd_HH_mm") + ".xls", stream.toByteArray(), subject, body);
```

Comment:
This method should be called sendReport, since the current name implies that an error report is being sent which is not the case